Add an option to specify the install prefix for CoreFoundation depend…

bryanpkc opened the pull request:
…encies.

Currently, swift-corelibs-xctest will fail to build if build-script option `--install-prefix` is set to use a location other than `/usr`. This is because build_script.py assumes that the CoreFoundation libraries can always be found under `$foundation_build_dir/usr/lib/swift`. This patch adds a new option to the Python script, to allow the top-level build-script to pass the correct prefix to the swift-corelibs-xctest build.

briancroom commented:
Cool, this seems to make sense. I find it a little strange that the *installation* prefix affects the layout of Foundation's *build products* directory, but I see that it does indeed work that way. 

Could you rebase this please? Sorry, I merged another PR which also had build script changes.

modocache commented on build_script.py:
```diff
@@ -307,13 +308,19 @@ def main(args=sys.argv[1:]):
+        "--foundation-install-prefix",
+        help="Path to the installation location for swift-corelibs-foundation "
+             "build products; CoreFoundation dependencies are expected to be "
+             "found under $foundation_build_dir/$foundation_install_prefix.",
```
I think it'd be nice to include in the description what the default value is. For example, the `--lit` argument does this: `"'%(default)s' by default."`.

modocache commented on build_script.py:
```diff
@@ -373,6 +380,12 @@ def main(args=sys.argv[1:]):
+        "--foundation-install-prefix",
+        help="Path to the installation location for swift-corelibs-foundation "
+             "build products; CoreFoundation dependencies are expected to be "
+             "found under $foundation_build_dir/$foundation_install_prefix.",
```
Same as above, I think it'd be nice to include what the `%(default)s` value is in this description.

bryanpkc commented:
@briancroom Thanks, I have rebased the patch.

@modocache Thank you for the suggestion. I have updated the help message to make it indicate the default value and look more consistent with the other messages.

briancroom commented:
@swift-ci please test

briancroom commented:
@swift-ci please test

briancroom commented:
@swift-ci please test

briancroom commented:
@swift-ci please test

briancroom commented:
CI seems unwilling to accept a request from me for this PR for whatever reason. The changes LGTM, but I'm unable to try it out myself right now and don't want to hit merge without seeing it actually execute. I'll try to give it a go tomorrow unless @modocache or someone else do so first.

modocache commented:
@swift-ci please test

modocache commented:
Swift CI appears to be a very fickle creature. It was listening to me just [two hours ago](https://github.com/apple/swift-corelibs-xctest/pull/112#issuecomment-219334610)...

modocache commented:
@swift-ci please test

modocache commented:
So strange! It listened to me [just a minute ago](https://github.com/apple/swift-corelibs-xctest/pull/113#issuecomment-219344857). @shahmishal, do you have any idea why @swift-ci is choosing when to obey orders?

![inline_space_odyssey](https://cloud.githubusercontent.com/assets/552921/15280156/22bd2070-1afb-11e6-8e8a-a4ab97a15cab.jpg)

Seriously, though, perhaps it has something to do with *who* submitted the pull request? I've noticed "@swift-ci please test" works for my own pull requests, but not this one.

bryanpkc commented:
Maybe it has to do with the repo being owned by an organization rather than a personal account? If that's the case, I can move the branch to a personal repo. But that would be sad because I have a bunch of other pull requests in the same boat.

briancroom commented:
Ok this checks out locally for me. I'm going to pull the trigger on it now.

modocache commented:
Great, thanks!

I only just noticed this is related to the https://github.com/apple/swift/pull/2541. Great work!! 💯

bryanpkc commented:
Thanks @briancroom!
